src/main/java/invtweaks/container/DirectContainerManager.java

```diff
@@ -82,8 +82,8 @@ public boolean move(ContainerSection srcSection, int srcIndex, ContainerSection
 
         // Use intermediate slot if we have to swap tools, maps, etc.
         assert srcStack != null;
-        if(destStack != null && srcStack.getItem() == destStack.getItem() && (srcStack.getMaxStackSize() == 1 ||
-                srcStack.hasTagCompound() || destStack.hasTagCompound())) {
+        if(destStack != null && srcStack.isItemEqual(destStack) && srcStack.getMaxStackSize() == 1 &&
+                ((srcStack.hasTagCompound() || destStack.hasTagCompound()) && (srcStack.getTagCompound().equals(destStack.getTagCompound())))) {
```

Comment:
I has tested this patch. if there are no check that the two item has same tag. the original condition will only swap the item which can stack. this patch will let them merge together.